### PR TITLE
Fix warnings from Visual Studio

### PR DESCRIPTION
**Pull request by chrisdembia**
### Brief summary of changes

This PR gets rid of the following warnings from MSVC:

```
C:\projects\opensim-core\OpenSim\Common\Test\testSTOFileAdapter.cpp(278): warning C4506: no definition for inline function 'const std::string OpenSim::DelimFileAdapter<double>::tableString(void)' [C:\projects\opensim_build\OpenSim\Common\Test\testSTOFileAdapter.vcxproj]
C:\projects\opensim-core\OpenSim\Simulation\Test\testProbes.cpp(623): warning C4506: no definition for inline function 'const std::string OpenSim::DelimFileAdapter<double>::tableString(void)' [C:\projects\opensim_build\OpenSim\Simulation\Test\testProbes.vcxproj]
C:\projects\opensim-core\OpenSim\Simulation\Test\testMuscleMetabolicsProbes.cpp(1414): warning C4506: no definition for inline function 'const std::string OpenSim::DelimFileAdapter<double>::tableString(void)' [C:\projects\opensim_build\OpenSim\Simulation\Test\testMuscleMetabolicsProbes.vcxproj]
C:\projects\opensim-core\OpenSim\Simulation\SimbodyEngine\Test\testConstraints.cpp(1125): warning C4506: no definition for inline function 'const std::string OpenSim::DelimFileAdapter<double>::tableString(void)' [C:\projects\opensim_build\OpenSim\Simulation\SimbodyEngine\Test\testConstraints.vcxproj]
...
C:\projects\opensim-core\Applications\Analyze\test\testAnalyzeTool.cpp(157): warning C4018: '<': signed/unsigned mismatch [C:\projects\opensim_build\Applications\Analyze\test\testAnalyzeTool.vcxproj]
...
C:\projects\opensim-core\OpenSim\Actuators\Test\testMuscles.cpp(968): warning C4018: '<': signed/unsigned mismatch [C:\projects\opensim_build\OpenSim\Actuators\Test\testMuscles.vcxproj]
C:\projects\opensim-core\OpenSim\Actuators\Test\testMuscles.cpp(979): warning C4101: 'x': unreferenced local variable [C:\projects\opensim_build\OpenSim\Actuators\Test\testMuscles.vcxproj]
```

### Testing I've completed

Ran API tests.

### CHANGELOG.md (choose one)

- no need to update because...these changes are not seen by users.

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

### CODE REVIEW

File: OpenSim/Common/Component.h [hunk 1]
```diff
@@ -2117,12 +2117,15 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     // End of System Creation and Access Methods.
     //@} 
 
-
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wunsupported-friend"
+#if defined(__clang__)
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wunsupported-friend"
+#endif
```
Comment (chrisdembia) [inline OpenSim/Common/Component.h]: Omitting the `#if defined (__clang__)` caused a `unknown pragma` warning with MSVC.

**Comment by chrisdembia** [comment]
Thanks @tkuchida

**Comment by aymanhab** [comment]
LGTM, merging

**Comment by chrisdembia** [comment]
Thanks @aymanhab

**Pull request merged**